### ISSUE TITLE
Resolve nonce issues with new mempool nonces from API?

### ISSUE THREAD

**Issue by Hero-Gamer** [blockstack/stacks-wallet#1193]
A. Sent 1 tx successfully with correct nonce
B. Tried sending 2nd tx afterwards, in preview window it says the same nonce as the 1st transaction, when it should be plus 1. 
Reinstalling to latest version doesn't fix it. 
Waiting minutes closing re-opening the desktop app doesn't fix it. 

What is the issue, has this issue always been in the app?

**Comment by Hero-Gamer** [comment]
about 15~20 minutes later, the 1st tx still "in mempool" spinning status, now if I attempt the 2nd tx the nonce is set correctly.

**Comment by 314159265359879** [comment]
Thanks for reporting Hero! I am seeing this too on testnet. The transaction has been confirmed in an anchorblock 5 hours ago yet still shows "in microblock". When I close and reopen the desktop app it does update here. The wallet should check for updates more often I believe. 

![image](https://github.com/hirosystems/desktop-wallet/assets/33360391/b8c8ad40-aafd-4c71-b579-c630fb69e036)

**Comment by markmhx** [comment]
This new API support for getting mempool nonces seems promising for improving our nonce calculations: https://github.com/hirosystems/stacks-blockchain-api/pull/1633

@fbwoolf wdyt since it appears you were the one that submitted the API-side issue that led to this PR last fall?

**Comment by fbwoolf** [comment]
> hirosystems/stacks-blockchain-api#1633

Yeah, I think we should try to use. We do it ourselves using pending transactions query, but would help to know what the api sends us.

**Comment by 314159265359879** [comment]
We get reminded now and again that this bug is still in the current wallet. 

It can take upto 5 minutes before the desktop wallet updates the next nonce after a transaction is send. On the extension this behavior is much better (takes less time to update).

**Comment by Hero-Gamer** [comment]
I waited more than 10 minutes this time before new tx is able to recognize the correct next nonce.

**Comment by 314159265359879** [comment]
Resolving this issue will not get prioritized yet. We have _revamping the desktop wallet_ on our roadmap for Q2 2024, this will likely have to wait until that time. 

Note that the team is focusing on improving support in hardware/Ledger mode and building the mobile version of Leather at the moment. Some of that work also involves making choices that benefit the desktop wallet later.
